File: Golang/2507-number-of-common-factors/number-of-common-factors.go
func commonFactors(a int, b int) int {
    small, count := a, 0
    if a > b {
        small = b
    }
    for i := 1; i <= small; i++ {
        if a % i == 0 && b % i == 0 {
            count++
        }
    }
    return count
}

File: Python/2507-number-of-common-factors/number-of-common-factors.py
class Solution:
    def commonFactors(self, a: int, b: int) -> int:
        minimum = min(a, b)
        factors = 0
        for i in range(1, minimum+1):
            if a % i == 0 and b % i == 0:
                factors += 1
        return factors